src/com/google/javascript/jscomp/AmbiguateProperties.java

```diff
@@ -87,7 +89,8 @@
   private final Map<String, Property> propertyMap = new HashMap<>();
 
   /** Property names that don't get renamed */
-  private final Set<String> externedNames;
+  private final Set<String> externedNames = new LinkedHashSet<>(
```

Comment:
@MatrixFrog suggested that this be an `ImmutableSet` instead, as long as you're changing things, so it's clear that it's not modified after the constructor. So this would be 

``` java
private final ImmutableSet<String> externedNames;
```